pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -144,7 +146,15 @@ public static SelectionResults renderSelectionResultsWithoutOrdering(@Nonnull Co
   public static List<String> getSelectionColumns(@Nonnull List<String> selectionColumns,
       @Nonnull IndexSegment indexSegment) {
     if (selectionColumns.size() == 1 && selectionColumns.get(0).equals("*")) {
-      List<String> allColumns = Arrays.asList(indexSegment.getColumnNames());
+      List<String> allColumns = new ArrayList<>();
+      for (String column: indexSegment.getColumnNames()) {
+        FieldSpec spec = indexSegment.getSegmentMetadata().getSchema().getFieldSpecFor(column);
+        if (spec instanceof MetricFieldSpec && ((MetricFieldSpec) spec).isDerivedMetric()) {
```

Comment:
@kishoreg  For the time-column related examples you brought up, do those columns need to be "derived" ? They can be regular columns in the metadata if we are not storing any association.